cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go

```diff
@@ -80,6 +80,9 @@ func (aws *awsCloudProvider) NodeGroups() []cloudprovider.NodeGroup {
 
 // NodeGroupForNode returns the node group for the given node.
 func (aws *awsCloudProvider) NodeGroupForNode(node *apiv1.Node) (cloudprovider.NodeGroup, error) {
+	if len(node.Spec.ProviderID) == 0 {
+		return nil, nil
```

Comment:
There should be at least an error-level log message here. Otherwise, the issues with missing ProviderId of nodes will be much harder to debug.

Reply:
I've added some logging, otherwise I agree this is just a quick workaround to solve the missing providerId problem. I think it would be useful warn about this problem but this should not affect other nodes in the node group if they have correct provdeirId's. Still reason of missing id's need to be discovered.